src/main/java/org/springframework/data/redis/connection/ReactiveStringCommands.java

```diff
@@ -184,6 +210,22 @@ default Mono<Boolean> set(ByteBuffer key, ByteBuffer value, Expiration expiratio
 				.map(BooleanResponse::getOutput);
 	}
 
+	/**
+	 * Set {@literal value} for {@literal key} with {@literal expiration} and {@literal options}. Return the old
+	 * string stored at key, or nil if key did not exist. An error is returned and SET aborted if the value
+	 * stored at key is not a string.
+	 *
+	 * @param key must not be {@literal null}.
+	 * @param value must not be {@literal null}.
+	 * @param expiration must not be {@literal null}. Use {@link Expiration#persistent()} for no expiration time or
+	 *          {@link Expiration#keepTtl()} to keep the existing.
+	 * @param option must not be {@literal null}.
+	 * @return
+	 * @see <a href="https://redis.io/commands/set">Redis Documentation: SET</a>
+	 * @since 3.4
+	 */
+	Flux<ByteBufferResponse<SetCommand>> setGet(ByteBuffer key, ByteBuffer value, Expiration expiration, SetOption option);
```

Comment:
Here we need both signatures, `Mono<ByteBuffer> setGet(ByteBuffer key, ByteBuffer value, Expiration expiration, SetOption option) {` and `Flux<ByteBufferResponse<SetCommand>> set(Publisher<SetCommand> commands);`

The first one delegates to the latter.